project-management/src/main/java/life/qbic/projectmanagement/application/measurement/MeasurementService.java

```diff
@@ -582,78 +454,126 @@ private List<MeasurementId> performRegistrationPxp(
+    var firstMetadataEntry = metadataList.get(0);
 
     var organisationQuery = organisationLookupService.organisation(
-        metadata.organisationId());
+        firstMetadataEntry.organisationId());
     if (organisationQuery.isEmpty()) {
       throw new MeasurementRegistrationException(ErrorCode.UNKNOWN_ORGANISATION_ROR_ID);
     }
 
-    var method = new ProteomicsMethodMetadata(instrumentQuery.get(), metadata.facility(),
-        metadata.fractionName(),
-        metadata.digestionMethod(), metadata.digestionEnzyme(),
-        metadata.enrichmentMethod(), Integer.parseInt(metadata.injectionVolume()),
-        metadata.lcColumn(), metadata.lcmsMethod());
+    var instrumentQuery = resolveOntologyCURI(firstMetadataEntry.instrumentCURI());
+    if (instrumentQuery.isEmpty()) {
+      throw new MeasurementRegistrationException(ErrorCode.UNKNOWN_ONTOLOGY_TERM);
+    }
 
-    var samplePreparation = new ProteomicsSamplePreparation(metadata.comment());
-    var labelingMethod = metadata.labeling().stream().map(label -> new ProteomicsLabeling(
-        label.sampleCode(), label.labelType(), label.label())).toList();
+    var method = new ProteomicsMethodMetadata(instrumentQuery.get(), firstMetadataEntry.facility(),
+        firstMetadataEntry.digestionMethod(), firstMetadataEntry.digestionEnzyme(),
+        firstMetadataEntry.enrichmentMethod(), firstMetadataEntry.lcColumn(),
+        firstMetadataEntry.lcmsMethod(), Integer.parseInt(firstMetadataEntry.injectionVolume()),
+        firstMetadataEntry.labeling()
+            .labelType());
 
-    var measurement = ProteomicsMeasurement.create(
-        projectId,
-        sampleIdCodeEntries.stream().map(SampleIdCodeEntry::sampleId).toList(),
-        selectedSampleCode,
-        organisationQuery.get(),
-        method, samplePreparation);
+    var measurement = ProteomicsMeasurement.create(projectId, assignedMeasurementCode,
+        organisationQuery.get(), method, specificMetadata);
 
-    metadata.assignedSamplePoolGroup()
-        .ifPresent(measurement::setSamplePoolGroup);
+    measurement.setSamplePoolGroup(firstMetadataEntry.samplePoolGroup());
 
-    measurement.setLabeling(labelingMethod);
+    return Map.of(measurement, sampleIdLookupTable.values());
+  }
 
-    measurement.setFraction(metadata.fractionName());
-    proteomicsMeasurements.put(measurement, sampleIdCodeEntries);
-    return proteomicsMeasurements;
+  private List<ProteomicsSpecificMeasurementMetadata> createSpecificMetadata(
+      List<ProteomicsMeasurementMetadata> metadata,
+      Map<SampleCode, SampleIdCodeEntry> sampleIdCodeLookupTable) {
+    return metadata.stream().map(metadataEntry -> ProteomicsSpecificMeasurementMetadata.create(
+        sampleIdCodeLookupTable.get(metadataEntry.associatedSample()).sampleId(),
+        metadataEntry.labeling().label(), metadataEntry.fractionName(),
+        metadataEntry.comment())).toList();
```

Comment:
Did you decide to keep the comment in? I thought you mentioned you wanted to remove the comment field.

Reply:
We need to go over in a separate PR when I talked to the stakeholders again. Need to check the mandatory fields, the templates, etc. too much for this PR